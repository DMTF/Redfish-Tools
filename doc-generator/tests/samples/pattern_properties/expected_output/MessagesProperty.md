| **Messages** { | object<br><br>* required* | The pattern property indicates that a free-form string is the unique identifier for the message within the registry. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**(pattern)** { | object | Property names follow regular expression pattern "\[A\-Za\-z0\-9\]\+" |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**ClearingLogic** { | object<br><br>*<br>(null)* | The clearing logic associated with this message.  The properties within indicate that what messages are cleared by this message as well as under what conditions. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**ClearsAll** | boolean<br><br>*read-only<br>(null)* | This property indicates that all prior conditions and messages are cleared provided the ClearsIf condition is met. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**ClearsIf** | string<br>(enum)<br><br>*read-only<br>(null)* | This property contains the available OEM specific actions for this resource. *See ClearsIf in Property Details, below, for the possible values of this property.* |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**ClearsMessage** [ ] | array (string, null)<br><br>*read-only* | This property contains the array of Message Ids that are cleared by this message, provided the other conditions are met. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;} |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Description** | string<br><br>*read-only required* | This is a short description of how and when this message is to be used. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Message** | string<br><br>*read-only required* | The actual message. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**NumberOfArgs** | integer<br><br>*read-only required* | The number of arguments to be expected to be passed in as MessageArgs for this message. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Oem** {} | object | Oem extension object. See the *Resource* schema for details on this property. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**ParamTypes** [ ] | array (string<br>(enum))<br><br>*read-only* | The MessageArg types, in order, for the message. *See ParamTypes in Property Details, below, for the possible values of this property.* |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Resolution** | string<br><br>*read-only required* | Used to provide suggestions on how to resolve the situation that caused the error. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Severity** | string<br><br>*read-only required* | This is the severity of the message. |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;} |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**(pattern)** {} [ ] | array, boolean, integer, number, object, string<br><br>*<br>(null)* | Property names follow regular expression pattern "^\(\[a\-zA\-Z\_\]\[a\-zA\-Z0\-9\_\]\*\)?@\(odata\|Redfish\|Message\)\\\.\[a\-zA\-Z\_\]\[a\-zA\-Z0\-9\_\.\]\+$" |
| } |   |   |
