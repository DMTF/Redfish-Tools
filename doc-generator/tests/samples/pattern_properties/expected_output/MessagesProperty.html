<tr><td><nobr><b>Messages</b> {</nobr></td><td>object</td><td> <nobr>required</nobr></td><td>The pattern property indicates that a free-form string is the unique identifier for the message within the registry.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>(pattern)</b> {</nobr></td><td>object</td><td></td><td>Property names follow regular expression pattern "[A-Za-z0-9]+"</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>ClearingLogic</b> <i>(v1.2+)</i> {</nobr></td><td>object</td><td> (null)</td><td>The clearing logic associated with this message.  The properties within indicate that what messages are cleared by this message as well as under what conditions.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>ClearsAll</b></nobr></td><td>boolean</td><td><nobr>read-only</nobr> (null)</td><td>This property indicates that all prior conditions and messages are cleared provided the ClearsIf condition is met.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>ClearsIf</b></nobr></td><td>string<br>(enum)</td><td><nobr>read-only</nobr> (null)</td><td>This property contains the available OEM specific actions for this resource.<br><i>See <a href="#redfish.dmtf.org/schemas/v1/MessageRegistry.json|details|ClearsIf">ClearsIf</a> in Property Details, below, for the possible values of this property.</i></td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>ClearsMessage</b> [ ] </nobr><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}</td><td>array<br>(string, null)</td><td><nobr>read-only</nobr></td><td>This property contains the array of Message Ids that are cleared by this message, provided the other conditions are met.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>Description</b></nobr></td><td>string</td><td><nobr>read-only</nobr> <nobr>required</nobr></td><td>This is a short description of how and when this message is to be used.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>Message</b></nobr></td><td>string</td><td><nobr>read-only</nobr> <nobr>required</nobr></td><td>The actual message.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>NumberOfArgs</b></nobr></td><td>integer</td><td><nobr>read-only</nobr> <nobr>required</nobr></td><td>The number of arguments to be expected to be passed in as MessageArgs for this message.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>Oem</b> { }</nobr></td><td>object</td><td></td><td>Oem extension object. See the <a href="http://redfish.dmtf.org/schemas/v1/Resource.json" target="_blank">Resource</a> schema for details on this property.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>ParamTypes</b> [ ] </nobr></td><td>array (string<br>(enum))</td><td><nobr>read-only</nobr></td><td>The MessageArg types, in order, for the message.<br><i>See <a href="#redfish.dmtf.org/schemas/v1/MessageRegistry.json|details|ParamTypes">ParamTypes</a> in Property Details, below, for the possible values of this property.</i></td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>Resolution</b></nobr></td><td>string</td><td><nobr>read-only</nobr> <nobr>required</nobr></td><td>Used to provide suggestions on how to resolve the situation that caused the error.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>Severity</b></nobr><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}</td><td>string</td><td><nobr>read-only</nobr> <nobr>required</nobr></td><td>This is the severity of the message.</td></tr>
<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<nobr><b>(pattern)</b> { } [ ] </nobr><br>}</td><td>array, boolean, integer, number, object, string</td><td> (null)</td><td>Property names follow regular expression pattern "^([a-zA-Z_][a-zA-Z0-9_]*)?@(odata|Redfish|Message)\.[a-zA-Z_][a-zA-Z0-9_.]+$"</td></tr>
